esp/public/media/scripts/ajaxschedulingmodule/ESP/Scheduler.js

```diff
@@ -17,176 +19,187 @@ function Scheduler(
+                var desc = resource_type.name;
+                if(resource.value) {
+                    desc += ': ' + resource.value;
+                }
+                room.resource_lines.push(desc);
             });
-            var desc = resource_type.name;
-            if(resource.value) {
-                desc += ': ' + resource.value;
+        });
+
+        $j.each(data.sections, function(index, section) {
+            requests = section.resource_requests[section.id];
+            if(requests) {
+                $j.each(requests, function(index, resource_array) {
+                    resource_array[0] = data.resource_types[resource_array[0]];
+                });
+            } else {
+                section.resource_requests = {};
+                section.resource_requests[section.id] = [];
             }
-            room.resource_lines.push(desc);
+
+
         });
-    });
 
-    $j.each(data.sections, function(index, section) {
-        requests = section.resource_requests[section.id];
-        if(requests) {
-            $j.each(requests, function(index, resource_array) {
-                resource_array[0] = data.resource_types[resource_array[0]];
-            });
+        this.lunch_timeslots = data.lunch_timeslots;
+        this.timeslots = new Timeslots(data.timeslots, this.lunch_timeslots);
+
+        this.rooms = data.rooms;
+
+        this.sections = new Sections(data.sections,
+                                     data.section_details,
+                                     data.categories,
+                                     data.teachers,
+                                     data.moderators,
+                                     data.schedule_assignments,
+                                     new ApiClient());
+
+        if(has_moderator_module === "True"){
+            this.moderatorDirectory = new ModeratorDirectory(moderatorEl,
+                                                             data.moderators);
         } else {
-            section.resource_requests = {};
-            section.resource_requests[section.id] = [];
+            this.moderatorDirectory = null;
         }
 
+        this.messagePanel = new MessagePanel(messageEl,
+                                             "Welcome to the Ajax Scheduler!<br><br>" +
+                                             "Delete: unschedule the selected section<br>" +
+                                             "Escape: unselect the selected section" + (has_moderator_module === "True" ? "/moderator": "") + "<br>" +
+                                             "F1: open the 'Classes' tab<br>" +
+                                             "F2: open the 'Room Filters' tab<br>" +
+                                             "F3: open the 'Checks' tab<br>" +
+                                             (has_moderator_module === "True" ? "F4: open the 'Moderators' tab<br>": "") +
+                                             "/: Search for a class");
```

Comment:
This is great. I think it could use a bit more styling. I might recommend bolding the key part, ie "**/**:", "**F3**:", etc. Or somehow differentiating the key part (maybe a more code-like font for just the keys) to make it nicer to read.